packages/app/pages/index.tsx

```diff
@@ -1,13 +1,197 @@
 import type { NextPage } from "next";
+import { useRouter } from "next/router";
 import Head from "next/head";
 import Link from "next/link";
-import { useCallback, useEffect, useRef, useState } from "react";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
 import Web3 from "web3";
 
 const DRAWER_WIDTH = 255;
 const HEADER_HEIGHT = 64;
 
+const Tab: React.FC<{ to: string }> = ({ children, to }) => {
+  const router = useRouter();
+  const onClick = useCallback(() => router.push(`#${to}`), [router, to]);
+  return (
+    <div
+      style={{ minHeight: 64, padding: 20, cursor: "pointer" }}
+      onClick={onClick}
+    >
+      {children}
+    </div>
+  );
+};
+
+const Portal: React.FC = ({ children }) => <>{children}</>; // Replace with react-portal in TOPO
+
+const Modal: React.FC<{
+  title: string;
+  isOpen: boolean;
+  setIsOpen: (b: boolean) => void;
+  confirmText?: string;
+  onConfirm?: () => void | Promise<void>;
+}> = ({
+  title,
+  isOpen,
+  setIsOpen,
+  children,
+  confirmText = "Submit",
+  onConfirm,
+}) => {
+  // TODO migrate to TOPO
```

Comment:
This whole component will be moved to TOPO. Should base components like this be created there first?